HeatStroke/StateMachine/StateMachine.cpp

```diff
@@ -165,7 +165,8 @@ void HeatStroke::StateMachine::Update(const float p_fDelta, const bool m_bUpdate
 	if (m_bUpdateStack)
 	{
 		// Update entire stack from bottom to top
-		StateStack::iterator it = m_mStateStack.begin(), end = m_mStateStack.end();
+		StateStack mLocalCopy(m_mStateStack);
+		StateStack::iterator it = mLocalCopy.begin(), end = mLocalCopy.end();
```

Comment:
That's because the mLocalCopy is copy-constructed, meaning it has the pointers to memory needed to iterate. The problem occurs if a State is pushed or popped from the stack while its iterating, then the iterators for m_mStateStack are reset and errors ensue. We need a local copy because it has the same information but different iterators.